pkg/build/api/helpers.go

```diff
@@ -75,3 +79,56 @@ func HasTriggerType(triggerType BuildTriggerType, bc *BuildConfig) bool {
 	matches := FindTriggerPolicy(triggerType, bc)
 	return len(matches) > 0
 }
+
+// RecordStageAndStepInfo records details about each build stage and step
+func RecordStageAndStepInfo(stages []StageInfo, stageName StageName, stepName StepName, startTime unversioned.Time, endTime unversioned.Time) ([]StageInfo, error) {
+	// If the stage already exists in the slice, update the DurationMilliseconds, and append the new step.
+	for stageKey, stageVal := range stages {
+		if stageVal.Name == stageName {
+			for _, step := range stages[stageKey].Steps {
+				if step.Name == stepName {
+					return stages, fmt.Errorf("error recording build timing information, step %v already exists in stage %v", stepName, stageName)
+				}
+			}
+			stages[stageKey].DurationMilliseconds = endTime.Time.Sub(stages[stageKey].StartTime.Time).Nanoseconds() / int64(time.Millisecond)
+			if len(stages[stageKey].Steps) == 0 {
+				stages[stageKey].Steps = make([]StepInfo, 0)
+			}
+			stages[stageKey].Steps = append(stages[stageKey].Steps, StepInfo{
+				Name:                 stepName,
+				StartTime:            startTime,
+				DurationMilliseconds: endTime.Time.Sub(startTime.Time).Nanoseconds() / int64(time.Millisecond),
+			})
+			return stages, nil
+		}
+	}
+
+	// If the stageName does not exist, add it to the slice along with the new step.
+	var steps []StepInfo
+	steps = append(steps, StepInfo{
+		Name:                 stepName,
+		StartTime:            startTime,
+		DurationMilliseconds: endTime.Time.Sub(startTime.Time).Nanoseconds() / int64(time.Millisecond),
+	})
+	stages = append(stages, StageInfo{
+		Name:                 stageName,
+		StartTime:            startTime,
+		DurationMilliseconds: endTime.Time.Sub(startTime.Time).Nanoseconds() / int64(time.Millisecond),
+		Steps:                steps,
+	})
+	return stages, nil
+}
+
+// AppendStageAndStepInfo appends the step info from one stages slice into another.
+func AppendStageAndStepInfo(stages []StageInfo, stagesToMerge []StageInfo) ([]StageInfo, error) {
```

Comment:
i'm not sure "it is possible" answers my question.  

1) is the code currently doing so?
2) if not, do you have an explicit scenario in mind where it would?